q2_qemistree/_hierarchy.py

```diff
@@ -8,58 +8,87 @@
 
 import biom
 import pandas as pd
-from sklearn.metrics import pairwise_distances
-from scipy.spatial.distance import squareform
+from scipy.spatial.distance import squareform, pdist
 from scipy.cluster.hierarchy import linkage
 from skbio import TreeNode
+import numpy as np
 from q2_feature_table import merge
 
 from ._collate_fingerprint import collate_fingerprint
-from ._match import match_label
+from ._match import match_tables
 from ._semantics import CSIDirFmt
 
 
-def build_tree(relabeled_fingerprints: pd.DataFrame) -> TreeNode:
+def pdist_union(u, v):
```

Comment:
Can you re-name this function to `_pdsit_union`.

Reply:
Can you also describe in the documentation that this function assumes the values are 0 and 1?